packages/verified-fetch/test/verified-fetch.spec.ts

```diff
@@ -310,13 +298,27 @@ describe('@helia/verifed-fetch', () => {
       })
     })
 
+    it('should return dag-json data with embedded CID', async () => {
+      const cid = CID.parse('baguqeerasords4njcts6vs7qvdjfcvgnume4hqohf65zsfguprqphs3icwea')
+      dagJsonStub.get.withArgs(cid).returns(Promise.resolve({
+        hello: 'world',
+        link: CID.parse('QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN')
+      }))
+      const resp = await verifiedFetch.fetch(cid)
+      const data = await resp.json()
+      expect(data).to.deep.equal({
+        hello: 'world',
+        link: CID.parse('QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN')
+      })
```

Comment:
This may be better, the CID encoded as a [link](https://ipld.io/specs/codecs/dag-json/spec/#links):

```suggestion
      expect(data).to.deep.equal({
        hello: 'world',
        link: {
          '/': 'QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN'
        }
      })
```

Reply:
I like this, along with https://github.com/ipfs/helia/pull/426#discussion_r1484525834

Reply:
Really naive question, but wouldn't the CID end up having two different object instances (even if it has the same underlying cid) causing the deep equal test to fail?

Reply:
Yeah, there would be two instances but the test would pass because deep equal compares every property of the two objects rather than asserting they are the same instance.

Reply:
`dag-json` is returned by gateways as `application/json`-compatible string. It has no concept of non-string representation of CID in JSON, and I don't think we should introduce a custom CID type in `.json()` response.

The way `.json()` works on the web is that it never adds custom types beyond standard JSON. If  `verified-fetch` aims to be a solid foundation to build on the web, we can't break that expectation. 

(+ usual bias: based on years of experience we had with JS ecosystem, I don't think we can confidently say that CID type introduced here won't break in a few months)

Please let's switch to string-only, as suggested by @achingbrain  above:

```js
expect(data).to.deep.equal({
        hello: 'world',
        link: {
          '/': 'QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN'
        }
})
```

If we want typed returns, that would could be introduced in the future as a separate abstraction, `.ipld()` or similar.